R/subsetting.R

```diff
@@ -661,6 +659,21 @@ fast_nrow <- function(x) {
   .row_names_info(x, 2L)
 }
 
+vectbl_assign <- function(x, i, value) {
+  if (is.logical(value)) {
+    if (.Call("tibble_need_coerce", value)) {
+      value <- vec_slice(x, NA_integer_)
+    }
+  } else {
+    if (.Call("tibble_need_coerce", x)) {
+      x <- vec_slice(value, rep(NA_integer_, length(x)))
+    }
+  }
+
+  vec_slice(x, i) <- value
```

Comment:
```suggestion
  x <- vec_assign(x, i, value)
```

re: https://github.com/r-lib/vctrs/issues/1409